test/behavior/bugs/16000.zig

```diff
@@ -0,0 +1,19 @@
+const expect = @import("std").testing.expect;
+
+const MyStruct = struct {
+    foobar: u8,
+};
+
+var my_struct1 = MyStruct{ .foobar = 1 };
+var my_struct2 = MyStruct{ .foobar = 2 };
+var my_struct3 = MyStruct{ .foobar = 3 };
+var my_struct4 = MyStruct{ .foobar = 4 };
+const foo: []const *MyStruct = &[_]*MyStruct{ &my_struct1, &my_struct2, &my_struct3, &my_struct4 };
+const bar: []const *MyStruct = foo[0..2] ++ foo[2..]; // this line will make compiler crash
+
+test {
+    try expect(bar[0].*.foobar == 1);
+    try expect(bar[1].*.foobar == 2);
+    try expect(bar[2].*.foobar == 3);
+    try expect(bar[3].*.foobar == 4);
+}
```

Comment:
In the future please move new tests into an already existing file in `test/behavior/` or create a new test case file in `test/cases/` with a name such as `ptr_arr_concat_something.zig`.
IIUC `test/behavior/bugs` is supposed to go away because we don't want GitHub linked with the Zig codebase and because issue numbers are undescriptive.

Reply:
thanks. I totally get the rationale for not using `test/behavior/bugs` anymore. Have I done it right this time?

Reply:
Yeah, I think it's better!